docs/user/openstack/vgpu.md

```diff
@@ -0,0 +1,456 @@
+display similar output:
+
+```console
+$ lspci -nn | grep -i nvidia
+3b:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 PCIe 16GB]
+[10de:1db4] (rev a1)
+```
+make sure to remove `nouveau` driver from loading. There might be necessary to
+add it to `/etc/modprobe.d/blacklist.conf` and/or changing
+grub config:
+
+```console
+$ sudo sed -i 's/console=/rd.driver.blacklist=nouveau console=/' /etc/default/grub
+$ sudo grub2-mkconfig -o /boot/grub2/grub.cfg
+```
+
+After that install host vGPU NVIDIA drivers:
+
+```console
+$ sudo rpm -iv NVIDIA-vGPU-rhel-8.4-510.73.06.x86_64.rpm
+```
+
+and reboot machine. Note, that drivers version may differ.
+
+After reboot, confirm there are correct drivers used:
+
+```console
+$ lsmod | grep nvidia
+nvidia_vgpu_vfio       57344  0
+nvidia              39055360  11
+mdev                   20480  2 vfio_mdev,nvidia_vgpu_vfio
+vfio                   36864  3 vfio_mdev,nvidia_vgpu_vfio,vfio_iommu_type1
+drm                   569344  4 drm_kms_helper,nvidia,mgag200
+```
+
+You can also use `nvidia-smi` tool for displaying device state.
+
+
+## OpenStack compute node
+
+There should be mediated devices populated by the driver (bus address may vary):
+
+```console
+$ ls /sys/class/mdev_bus/0000\:3b\:00.0/mdev_supported_types/
+nvidia-105  nvidia-106  nvidia-107  nvidia-108  nvidia-109  nvidia-110
+nvidia-111  nvidia-112  nvidia-113  nvidia-114  nvidia-115  nvidia-163
+nvidia-217  nvidia-247  nvidia-299  nvidia-300  nvidia-301
+```
+
+Depending of the type of workload and purchased license edition, appropriate
+types needs to be configured in `nova.conf` for compute node, i.e.:
+
+```ini
+...
+[devices]
+enabled_vgpu_types: nvidia-105
+
+...
+```
+
+After compute service restart, placement-api should report additional resources -
+command `openstack resource provider list` and `openstack resource provider inventory list <id of the main provider>`
+should display VGPU resource class available. For more information
+[navigate to OpenStack Nova docs](https://docs.openstack.org/nova/latest/admin/virtual-gpu.html).
+
+
+## OpenStack vGPU flavor
+
+Now, create a flavor, to be used to spin up new vGPU enabled nodes:
+
+```console
+$ openstack flavor create --disk 25 --ram 8192 --vcpus 4 \
+    --property "resources:VGPU=1" --public <nova_gpu_flavor>
+```
+
+
+## Create vGPU enabled Worker Nodes
+
+Worker nodes can be created by using machine API. To do that,
+[create new machineSet in OpenShift](https://docs.openshift.com/container-platform/4.11/machine_management/creating_machinesets/creating-machineset-osp.html).
+
+```console
+$ oc get machineset -n openshift-machine-api <machineset_name> -o yaml > vgpu_machineset.yaml
+```
+
+Edit yaml file, be sure to have different name, have replicas set to the amount
+of your cGPU capacity at maximum and set the right flavor, which would hint
+OpenStack about right resources to include into virtual machine:
```

Comment:
I would make sure to say the below YAML is just an example, user's YAMLs might be different.